ui/pages/confirmations/components/confirm/info/hooks/useBatchApproveBalanceChanges.ts

```diff
@@ -66,7 +70,7 @@ function useBatchApproveSimulationBalanceChanges({
 }) {
   return useAsyncResult(
     async () => buildSimulationTokenBalanceChanges({ nestedTransactions }),
-    [nestedTransactions],
+    [nestedTransactions?.length],
```

Comment:
If we're using the full `nestedTransactions` in the callback, then we have no choice but to use it as a dependency also, as otherwise we won't re-query with the latest data?

For example, if it was entirely new, but the same length.

Reply:
`nestedTransactions` are not likely to change thus even `nestedTransactions?.length` is ok to depend on and this does seem to fix re-rendering issue.

Reply:
Unfortunately we can't couple that decision here though as this component shouldn't know how often `nestedTransactions` will update, only that we do a request when it changes.

If this is the fix, that suggests the deeper issue is that the `nestedTransactions` reference is changing despite the data itself not changing.

So ideally we'd debug that or as a temporary fix only, use a "deep equal" dependency such as `JSON.stringify(nestedTransactions)`.